### ISSUE TITLE
Option "Send to active instrument track" preserves existing FX-plugins

### ISSUE THREAD

**Issue by musikBear** [LMMS/lmms#3419]
This may be intended?
OS  : winXP sp3
LMMS 1.1.90-g33d8c44

**Issue description:**
If the sidebar option *"Send to active instrument track"* is used, the existing preset (the active) will not be completely overwritten. The existing plugins in FX-Tab is preserved in the 'new' preset, and it is kind of a 'hybrid' between the existing and the inserted

**Scope**
zasfx, AFP, & Kicker will preserve inserted FX, all the others overwrites the FX content.

**Recreate         :**
Open an empty project
Insert an zasfx empty instrument
Add any effect on the FX-tab. (i just took dual filter)
Keep FX-tab open
Open the zasfx presets
Right-click on Pads | sine-pad0001 
Use option "Send to active instrument track"
The preset sine-pad0001 is inserted in the opend/active preset, but the filter in the FX-tab is preserved.
I also tried with 3oc. Those presets are completely replaced, and an 3oc preset, will also completely replace a zasfx preset. That is not the case for zasfx

(That was how i messed up in 3415.

**Comment by tresf** [comment]
> No description provided.

Closed.

**Comment by musikBear** [comment]
Dont understand -is this already fixed in Master?

**Comment by tresf** [comment]
So you added effects to an instrument, told that instrument to use a new preset and you're upset that it kept the effects that you explicitly added?

Why would you send to the active instrument if you didn't want the settings?

If you don't want effects preserved, state that as a new feature.  The existing behavior has been the behavior for a very long time.

**Comment by musikBear** [comment]
@tresf I have now tested all instruments for this behaviour. 
zasfx, AFP, & Kicker will preserve inserted FX, all the others overwrites the FX content.
Imo all instruments should have same behaviour.
I did not test Vestige, because using 'send to active' with VSTs crashes my pc/ lmms
I have only tested with one effect, but i doubt that would differ, i did try one other, and the result was the same.

**Comment by BaraMGB** [comment]
If this is true I would say this is a bug. But I'm not sure which behavior is the bug and which is the feature. @tresf in my opinion all presets should erase the previous effects. 

Imagine you want to hear your composition with different presets. Therefore you drop them successively to the channel. But if the effects are preserved on that channel this work flow isn't possible.

**Comment by tresf** [comment]
@BaraMGB agreed, I had [requested @musikBear fix his bug report](https://github.com/LMMS/lmms/issues/3415#issuecomment-286118319) prior to reopening.

In regards to intended behavior... For presets, this make sense, I agree, however... for sample, this is not necessarily a good idea.

For example:
 * Take 3 recordings of sound (e.g. lyrics for a song)
 * Use first recording for adjusting volume, add compression, EQ
 * Decide second recording is better for track, send to active instrument
 * Lose all presets associated with recording

One could argue that this should all be done through the FX mixer and although I agree, we don't have an easy way to save an FX preset and we also don't have a way to migrate an FX preset from the Instrument to a Mixer channel.

I'm a bit torn on the decision, personally.  I've used the "bug" on both sides of the fence to satisfaction.  I think I would lean towards:
 * Keep FX if sample/patch/vst.  Samples/plugins affected:  afp, sf2, gig, vst, patman (.sf2, .gig, .wav, .ogg, .flac, .dll, .pat) and anything else I missed.
 * Drop FX if preset.  Presets affected: `xpf`, `xiz`, and I think `xml` is technically supported too.

We could also consider making all new items wipe FX by design and have a context menu to preserve FX.

**Comment by tresf** [comment]
Additionally... there's a slight discrepancy between dragging a sample/patch/bank over an instrument button versus doing the same with a preset file.  The behavior should be identical.
 * Dragging a preset file over an instrument button will replace that instrument with the preset
 * Dragging a sample file over an instrument button will add a NEW TRACK to the bottom of the song editor

**Comment by BaraMGB** [comment]
>We could also consider making all new items wipe FX by design and have a context menu to preserve FX.

For me that seems the cleanest way for this.

**Comment by tresf** [comment]
If it's any help... I personally use the AFP dialog to replace samples, so the use-case I mentioned I'm not at all invested in, I just wanted to offer it as a valid use-case.
